src/components/Editor/Reducers/loadProjectReducers.js

```diff
@@ -0,0 +1,57 @@
+const loadProjectPending = (state, action) => {
+  state.loading = "pending";
+  state.accessDeniedNoAuthModalShowing = false;
+  state.modals = {};
+  state.currentLoadingRequestId = action.meta.requestId;
+};
+
+const loadProjectFulfilled = (state, action) => {
+  if (
+    state.loading === "pending" &&
+    state.currentLoadingRequestId === action.meta.requestId
+  ) {
+    state.project = action.payload.project;
+    state.loading = "success";
+    state.justLoaded = true;
+    state.saving = "idle";
+    state.currentLoadingRequestId = undefined;
+    if (state.openFiles.flat().length === 0) {
+      const firstPanelIndex = 0;
+      if (state.project.project_type === "html") {
+        state.openFiles[firstPanelIndex].push("index.html");
+      } else {
+        state.openFiles[firstPanelIndex].push("main.py");
+      }
+    }
+  }
+};
+
+const loadProjectRejected = (state, action) => {
+  if (
+    state.loading === "pending" &&
+    state.currentLoadingRequestId === action.meta.requestId
+  ) {
+    state.loading = "failed";
+    state.saving = "idle";
+    const splitErrorMessage = action.error.message.split(" ");
+    const errorCode = splitErrorMessage[splitErrorMessage.length - 1];
+    const accessToken = action.meta.arg?.accessToken;
+    const accessDeniedCodes = ["401", "403", "500"];
+
+    if (errorCode === "404") {
+      state.notFoundModalShowing = true;
+    } else if (accessDeniedCodes.includes(errorCode) && accessToken) {
```

Comment:
🧐 I suspect this isn't working as expected. Wondering if `accessToken` is an object regardless?